src/Acuminator/Acuminator.Utils/RoslynExtensions/PXFieldAttributes/AttributeInformation.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Acuminator.Analyzers;
+using Microsoft.CodeAnalysis;
+
+namespace Acuminator.Utilities
+{
+	/// <summary>
+	/// Information about the Acumatica field attributes.
+	/// </summary>
+	public class AttributeInformation
+	{
+		private readonly PXContext _context;
+
+
+		public AttributeInformation(PXContext pxContext)
+		{
+			pxContext.ThrowOnNull(nameof(pxContext));
+
+			_context = pxContext;
+		}
+
+		public bool ContainsBaseType(ITypeSymbol attributeSymbol, ITypeSymbol type)
+		{
+			attributeSymbol.ThrowOnNull(nameof(attributeSymbol));
+			type.ThrowOnNull(nameof(type));
+
+			List<ITypeSymbol> attributeTypeHierarchy = attributeSymbol.GetBaseTypesAndThis().ToList();
+			if (attributeTypeHierarchy.Contains(type))
+				return true;
+			return false;
+		}
+
+		public bool AttributeDerivedFromClass(ITypeSymbol attributeSymbol, ITypeSymbol type, int depth = 10)
+		{
+			attributeSymbol.ThrowOnNull(nameof(attributeSymbol));
+			type.ThrowOnNull(nameof(type));
+			if (depth <= 0 || depth > 100)
+				return false;
+			if (ContainsBaseType(attributeSymbol, type))
+				return true;
+
+			//get symbols from PXContext
+			//PX.Data.PXAggregateAttribute and PX.Data.PXDynamicAggregateAttribute
+			var aggregateAttribute = _context.AttributeTypes.PXAggregateAttribute;
+			var dynamicAggregateAttribute = _context.AttributeTypes.PXDynamicAggregateAttribute;
+
+			if (ContainsBaseType(attributeSymbol, aggregateAttribute) || ContainsBaseType(attributeSymbol, dynamicAggregateAttribute))
+			{
+				var allAttributes = attributeSymbol.GetAllAttributesDefinedOnThisAndBaseTypes().ToList();
+				foreach (var attribute in allAttributes)
+				{
+					//go in recursuion
+					var result = AttributeDerivedFromClass(attribute, type, --depth);
+					if (depth <= 0 || depth > 100)
+						return false;
+					if (result)
+						return result;
+				}
+			}
+			return false;
+		}
+
+		public bool IsBoundAttribute(ITypeSymbol attributeSymbol)
+		{
+			//Get this symbols from PXContext
+			//PX.Data.PXDBFieldAttribute
+			var dbFieldAttribute = _context.FieldAttributes.PXDBFieldAttribute;
+			return AttributeDerivedFromClass(attributeSymbol, dbFieldAttribute);
```

Comment:
@saloponov  the check if attribute is derived from PXDBFieldAttribute can't be used to check if the attribute is DB bound. 
@dropsonic  correct me if I'm wrong here, but there are PXDBUserPasswordAttribute and PXDBCalcedAttribute which do not derive from the PXDBFieldAttribute and don't have any aggregator attributes in inheritance chain which are marked with PXDBFieldAttribute

Reply:
@SENya1990 I think that this is a mistake in Acumatica code. `PXDBScalarAttribute` is similar to `PXDBCalcedAttribute`, but it inherits from `PXDBFieldAttribute` and it is marked with `[PXAttributeFamily(typeof(PXDBFieldAttribute))]`.
I suggest to check if an attribute inherits from either `PXDBFieldAttribute` or `PXDBCalcedAttribute`. I don't think there is another way to determine it.

Reply:
@dropsonic actually there is a register of attributes which stores sets of bound and unbound field attributes, It can be used here, for code re-use. I wrote a small utility which retrieved attributes declared in Acumatica and added field attributes to register.  The algorithm checks if register contains any type from attribute's inheritance chain. 
Therefore, because bound field attributes set contains PXDBFieldAttribute, all attributes derived from it will be recognised as bound.